### PR TITLE
Removes redundant buffer zeroing in JSONSerialization

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test

File: Sources/Foundation/JSONSerialization.swift [hunk 1]
```diff
@@ -209,14 +209,14 @@ open class JSONSerialization : NSObject {
...
         repeat {
-            var buffer = [UInt8](repeating: 0, count: 1024)
-            var bytesRead: Int = 0
-            bytesRead = stream.read(&buffer, maxLength: buffer.count)
-            if bytesRead < 0 {
-                throw stream.streamError!
-            } else {
-                data.append(&buffer, count: bytesRead)
+            let buffer = try [UInt8].init(unsafeUninitializedCapacity: 1024) { buf, initializedCount in
+                let bytesRead = stream.read(buf.baseAddress!, maxLength: 1024)
```
Comment (spevans) [inline Sources/Foundation/JSONSerialization.swift]: Could the second `1024` be changed to `buf.count` just to remove the duplicated constant.

Comment (valeriyvan) [reply Sources/Foundation/JSONSerialization.swift]: Fixed

**Comment by spevans** [comment]
@swift-ci test linux

**Pull request merged**